store/src/sql/sqlite/event.rs

```diff
@@ -44,16 +66,22 @@ impl SqliteEventStore {
         let max = max_delivered
             .res
             .checked_add(1)
-            .context("More than i64::MAX delivered events!")?;
-
-        GLOBAL_COUNTER.fetch_max(max, std::sync::atomic::Ordering::SeqCst);
-
+            .ok_or_else(|| Error::new_fatal(anyhow!("More than i64::MAX delivered events!")))?;
+        if let Some(ref t) = self.test_counter {
+            t.fetch_max(max, std::sync::atomic::Ordering::SeqCst);
+        } else {
+            GLOBAL_COUNTER.fetch_max(max, std::sync::atomic::Ordering::SeqCst);
+        }
         Ok(())
     }
 
     // could change this to rely on time similar to a snowflake ID
```

Comment:
Yes, looking at this, it needs to be a snowflake, since we can run out of events and will continue to error at that point.

Reply:
Maybe? I thought snowflakes were still limited to u64 items (and typically less since it uses some bits for node ID). This isn't intended to be globally unique and is just a value we set in our local sqlite DB to make sure we return them in the right order. Right now, the sqlite table will break if we go over u64 (integer column). Having a time component could be useful but I'm not sure a snowflake or similar is necessary.